### PR TITLE
Remove v1 features and implement new product naming scheme

### PR DESCRIPTION
**Pull request by asjohnston-asf**
### Deprecate v1

Revises all logging to use a module logger.  Command line scripts now log to STDOUT.

Removes the v1 entrypoint from `__main__.py`.  v2 is now the only entrypoint, invoked via `insar_gamma <args>`

Removes `ifm_sentinel.py` parameters not used in v2:
- `output` Output igram directory
- `--dem` Input DEM file to use
- `-i` Create incidence angle file
- `-o` Use opentopo to get the DEM file instead of get_dem
- `-c` cross pol processing - either hv or vh (default hh or vv)
- `-t` Start processing at time for length bursts
- `-m` Apply water body mask to DEM file prior to processing

Removes unused stacking support in `stack_sentinel.py`

Removes `par_s1_slc.py` in favor of duplicate functionality offered by `hyp3lib.par_s1_slc_single`

### V2 Features added

Implements new [product naming scheme](https://docs.google.com/spreadsheets/d/1Rzla4Uy7Av6UJ_glEA85WRtI881g_Mj8UYZ2Kex0-f8/edit#gid=879833876)

### CODE REVIEW

**Comment by khogenso** [comment]
This approach appears to lock the repo into being a hyp3 plug-in, planning to eliminate direct use.  Some of those options we're pulling are useful when doing custom processing not in hyp3

File: conda-env.yml [hunk 1]
```diff
@@ -13,16 +13,13 @@ dependencies:
...
   - lxml
-  - pillow
   - pip:
     # for packaging and testing
     - s3pypi
     - safety
     # For running
     - --trusted-host hyp3-pypi.s3-website-us-east-1.amazonaws.com
       --extra-index-url http://hyp3-pypi.s3-website-us-east-1.amazonaws.com
-    - hyp3proclib>=1.0.1,<2
```
Comment (jhkennedy) [inline conda-env.yml:28]: :partying_face:

File: hyp3_insar_gamma/ifm_sentinel.py [hunk 2]
```diff
@@ -1,38 +1,31 @@
...
-from hyp3_insar_gamma.getDemFileGamma import getDemFileGamma
+from hyp3_insar_gamma.getDemFileGamma import get_dem_file_gamma
 from hyp3_insar_gamma.interf_pwr_s1_lt_tops_proc import interf_pwr_s1_lt_tops_proc
-from hyp3_insar_gamma.par_s1_slc import par_s1_slc
 from hyp3_insar_gamma.unwrapping_geocoding import unwrapping_geocoding
 
-# FIXME: refactor to eliminate globals
-global lasttime
-global log
-global proc_log
```
Comment (jhkennedy) [inline hyp3_insar_gamma/ifm_sentinel.py:26]: :partying_face:

File: hyp3_insar_gamma/ifm_sentinel.py [hunk 3]
```diff
@@ -426,31 +355,17 @@ def main():
...
     parser.add_argument("reference", help="Reference input file")
     parser.add_argument("secondary", help="Secondary input file")
-    parser.add_argument("output", help="Output igram directory")
-    parser.add_argument("-d", "--dem",
-                        help="Input DEM file to use, otherwise calculate a bounding box (e.g. big for big.dem/big.par)")
     parser.add_argument("-r", "--rlooks", default=20, help="Number of range looks (def=20)")
     parser.add_argument("-a", "--alooks", default=4, help="Number of azimuth looks (def=4)")
-    parser.add_argument("-i", action="store_true", help="Create incidence angle file")
     parser.add_argument("-l", action="store_true", help="Create look vector theta and phi files")
     parser.add_argument("-s", action="store_true", help="Create line of sight displacement file")
```
Comment (jhkennedy) [inline hyp3_insar_gamma/ifm_sentinel.py:361]: Both of these should probably have a long variable name

**Review by jhkennedy — COMMENTED** [review]
We could also PEP8-ify the module/file names here as well, but we don't *need* to yet

**Review by Jlrine2 — APPROVED** [review]

**Pull request merged**